connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java

```diff
@@ -0,0 +1,335 @@
+                }
+            });
+        }
+    }
+
+    /* default */ static Connector configureConnector(final ConnectorTemplate connectorTemplate, final Connector connector,
+        final String specification) {
+
+        final Connector.Builder builder = new Connector.Builder().createFrom(connector);
+
+        final Swagger swagger = new SwaggerParser().parse(specification);
+        addGlobalParameters(builder, swagger);
+
+        final Map<String, Path> paths = swagger.getPaths();
+
+        final String connectorId = connector.getId().get();
+        final String connectorGav = connectorTemplate.getCamelConnectorGAV();
+        final String connectorScheme = connectorTemplate.getCamelConnectorPrefix();
+
+        int idx = 0;
+        for (final Entry<String, Path> pathEntry : paths.entrySet()) {
+            final Path path = pathEntry.getValue();
+
+            final Map<HttpMethod, Operation> operationMap = path.getOperationMap();
+
+            for (final Entry<HttpMethod, Operation> entry : operationMap.entrySet()) {
+                final Operation operation = entry.getValue();
+                if (operation.getOperationId() == null) {
+                    operation.operationId("operation-" + idx++);
+                }
+
+                final ActionDefinition actionDefinition = createActionDefinition(specification, operation);
+
+                final Action action = new Action.Builder()//
+                    .id(createActionId(connectorId, connectorGav, operation))//
+                    .name(Optional.ofNullable(operation.getSummary())
+                        .orElseGet(() -> entry.getKey() + " " + pathEntry.getKey()))//
+                    .description(Optional.ofNullable(operation.getDescription()).orElse(""))//
+                    .camelConnectorGAV(connectorGav)//
+                    .camelConnectorPrefix(connectorScheme)//
+                    .connectorId(connectorId)//
+                    .tags(Optional.ofNullable(operation.getTags()).orElse(Collections.emptyList()))//
+                    .definition(actionDefinition)//
+                    .build();
+
+                builder.addAction(action);
+            }
+        }
+
+        if (idx != 0) {
+            // we changed the Swagger specification by adding missing
+            // operationIds
+            builder.putConfiguredProperty("specification", serialize(swagger));
+        }
+
+        return builder.build();
+    }
+
+    /* default */ static ActionDefinition createActionDefinition(final String specification,
+        final Operation operation) {
+        final ActionDefinition.Builder actionDefinition = new ActionDefinition.Builder();
+
+        final Optional<BodyParameter> maybeRequestBody = operation.getParameters().stream()
+            .filter(p -> p instanceof BodyParameter && ((BodyParameter) p).getSchema() != null)
+            .map(BodyParameter.class::cast).findFirst();
+        if (maybeRequestBody.isPresent()) {
```

Comment:
Could be rewritten as (something like anyway!):

```java
actionDefinition.inputDataShape(
    maybeRequestBody
        .flatMap((requestBody) -> createShapeFromModel(specification, requestBody.getSchema()))
        .orElse(DATA_SHAPE_NONE)
);
```

Forgive me for playing with `Optional`s - not 100% sure which is more readable tbh...